### ISSUE TITLE
Unable to launch chrome in win10 

### ISSUE THREAD

**Issue by Tribhuvanaucse** [selenium-cucumber/selenium-cucumber-ruby#33]
Hey Sameer,
I am using Rspec as directed by my company . and also I follow your steps of understanding how we use the Selenium webdriver with ruby. 

I have succesfully setup the test automation and working fine in Firefox, 

When I try to execute the same in chrome and IE , the browser doesnt get launch and give me this error.

Failure/Error: @driver = Selenium::WebDriver.for :chrome

     Selenium::WebDriver::Error::WebDriverError:
       unable to connect to chromedriver 127.0.0.1:9515

**Comment by sameer49** [comment]
@Tribhuvanaucse 
is chromdriver is in system path ?
what is selenium and chrome version that  your using ?

**Comment by Tribhuvanaucse** [comment]
Yes Sameer , I followed the same instruction mentioned in your video . i
have set the path for chrome and Ie driver .

Selenium i am using 3.4.0
chrome browser version is 58

let me know if I am missing anything

**Comment by sameer49** [comment]
what is chromedriver version?

**Comment by Tribhuvanaucse** [comment]
Its 2.29

http://chromedriver.storage.googleapis.com/index.html?path=2.29/

i am using the file chromedriver_win32.zip
<http://chromedriver.storage.googleapis.com/2.29/chromedriver_win32.zip>

Thanks

**Comment by sameer49** [comment]
ok
open command prompt and enter following two commands and post here whats output is getting 
command 1 : "chromedriver"
command 2 : netstat -a -n -o | find "9515"

**Comment by Tribhuvanaucse** [comment]
sameer here is the result:


command 1(  check the screenshot 1)

command 2: nothing happen but it get executed without any error

**Comment by sameer49** [comment]
can't see screenshot attached

**Comment by Tribhuvanaucse** [comment]
can i have your email id?

**Comment by sameer49** [comment]
just paste the output

**Comment by Tribhuvanaucse** [comment]
sameer there is no output comining .. when i type chromediver , the windows close the chrome browser automatically and for the second command nothing is displayed in comanad prompt

**Comment by Tribhuvanaucse** [comment]
![problem in chrome](https://cloud.githubusercontent.com/assets/12486089/26065644/e844fe54-3994-11e7-98c0-7574d52d7baf.JPG)
![second comand](https://cloud.githubusercontent.com/assets/12486089/26065652/eb930c86-3994-11e7-8cfd-a92901cca5a6.JPG)

**Comment by sameer49** [comment]
ok hit "set" command and send its output, please don't send screenshot.

**Comment by Tribhuvanaucse** [comment]
C:\webdrivers>set
ALLUSERSPROFILE=C:\ProgramData
ANSICON=120x9001 (120x30)
ANSICON_DEF=7
APPDATA=C:\Users\BL-Team\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=DESKTOP-9OIMJGI
ComSpec=C:\WINDOWS\system32\cmd.exe
FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer
FPS_BROWSER_USER_PROFILE_STRING=Default
HOMEDRIVE=C:
HOMEPATH=\Users\BL-Team
LOCALAPPDATA=C:\Users\BL-Team\AppData\Local
LOGONSERVER=\\DESKTOP-9OIMJGI
NUMBER_OF_PROCESSORS=4
OneDrive=C:\Users\BL-Team\OneDrive
OS=Windows_NT
Path=C:\ProgramData\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\webdrivers;C:\Program
Files
(x86)\Skype\Phone\;C:\Ruby23-x64\bin;C:\Users\BL-Team\AppData\Local\Microsoft\WindowsApps
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.RB;.RBW
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=3d04
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Program
Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
PUBLIC=C:\Users\Public
SESSIONNAME=Console
SystemDrive=C:
SystemRoot=C:\WINDOWS
TEMP=C:\Users\BL-Team\AppData\Local\Temp
TMP=C:\Users\BL-Team\AppData\Local\Temp
USERDOMAIN=DESKTOP-9OIMJGI
USERDOMAIN_ROAMINGPROFILE=DESKTOP-9OIMJGI
USERNAME=BL-Team
USERPROFILE=C:\Users\BL-Team
windir=C:\WINDOWS

**Comment by sameer49** [comment]
is chrome browser (not driver)  32 bit or 64bit ?

https://www.howtogeek.com/199307/how-to-tell-if-you-have-the-32-bit-or-64-bit-version-of-google-chrome/

**Comment by Tribhuvanaucse** [comment]
64bit sameer

**Comment by sameer49** [comment]
here is the issue, you said your using 32bit driver change it to 64bit

**Comment by Tribhuvanaucse** [comment]
but the question is , when i go the chromedriver , i dont see 64 bit for
window...

**Comment by sameer49** [comment]
sorry for windows 32bit driver supported, and it works for bot 32bit and 64bit.
try to change chrome browser to 32bit version

**Comment by Tribhuvanaucse** [comment]
sameer thank you.
i upgraded to 32bit google chrome browser  but when I launch the browser
its get automatically crash, its the same which i attached the
screenshots... b

unfortunately when i run the test scripts , the browser is not launching ..


:(

**Comment by sandeepnagra** [comment]
Hi Tribhuvanaucse,

I also tried several versions of Chrome driver but eventually I had to downgrade selenium-web driver to 3.2.1 to make it work with driver 2.29. I do have 64-bit chrome browser with version 58.

**Comment by Tribhuvanaucse** [comment]
I guess then i will follow you .. but i will upgrade to chrome browser
64bit as i have seen that many people are saying that chromediver works for
both 32 and 64 bith cjhrome browser

**Comment by sandeepnagra** [comment]
Yes, it does. I have been using Chrome driver with 64-bit chrome for a long time now.

**Comment by Tribhuvanaucse** [comment]
one more question
when how we can downgrade ?
bcoz when I type gem install selenium-webdriver, it will install the latest
one right?

correct me if i am missing anything?

**Comment by sameer49** [comment]
@tribhuvanaucse

When you hit command chromedriver it should start on port 9515
In your case it's trying to  lunch chrome.exe which is should not happen

**Comment by Tribhuvanaucse** [comment]
i followed the same steps mentioned in your video ..

**Comment by sandeepnagra** [comment]
Just uninstall current version and the reinstall using gem install selenium-webdriver  -v 3.2.1

**Comment by sameer49** [comment]
No, its not issue with gem version
when chromedriver command is enter it should give following output : 

```
C:\Users\sameersawant222gmail>chromedriver
Starting ChromeDriver 2.29.461591 (62ebf098771772160f391d75e589dc567915b233) on
port 9515
Only local connections are allowed
```
In this case its crashing
so try different chromedriver versions

**Comment by Tribhuvanaucse** [comment]
@sandeep, thank you for the suggestion, unfortunately i did but still
unable to launch the chrome.

**Comment by Tribhuvanaucse** [comment]
@sameer, i am not able to figure out the problem  that what could be the
root cause for this.

but when I use the Java IDE and tried to automate with cucumber then it
runs perfectly..
